src/hera/workflows/_mixins.py

```diff
@@ -575,20 +575,23 @@ def _build_arguments(self) -> Optional[ModelArguments]:
             if isinstance(arg, dict):
                 for k, v in arg.items():
                     value = Parameter(name=k, value=v)
-                    result.parameters = [value] if result.parameters is None else result.parameters + [value]
+                    result.parameters = (
+                        [value.as_argument()]
+                        if result.parameters is None
+                        else result.parameters + [value.as_argument()]
+                    )
             elif isinstance(arg, ModelArtifact):
                 result.artifacts = [arg] if result.artifacts is None else result.artifacts + [arg]
             elif isinstance(arg, Artifact):
                 result.artifacts = (
                     [arg._build_artifact()] if result.artifacts is None else result.artifacts + [arg._build_artifact()]
                 )
-            elif isinstance(arg, ModelParameter):
-                result.parameters = [arg] if result.parameters is None else result.parameters + [arg]
             elif isinstance(arg, Parameter):
                 result.parameters = (
                     [arg.as_argument()] if result.parameters is None else result.parameters + [arg.as_argument()]
                 )
-
+            elif isinstance(arg, ModelParameter):
```

Comment:
Does `ModelArtifact` also need moving after `Artifact`?

Reply:
Interestingly, no. That's because `Artifact` inherits from `BaseModel` not `ModelArtifact`